tools/src/main/java/org/terracotta/utilities/logging/LoggingOutputStream.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2020 Terracotta, Inc., a Software AG company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.utilities.logging;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import org.slf4j.Logger;
+import org.slf4j.event.Level;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.WrongMethodTypeException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Implements an {@code OutputStream} that forwards lines written to it to
+ * a {@link Logger} at a specified level.  Bytes are accumulated by this
+ * {@code OutputStream} until a {@link System#lineSeparator()} value is found;
+ * once the line ending is found, the line is flushed to the logger.
+ */
+public class LoggingOutputStream extends OutputStream {
+
+  public static final int DEFAULT_BUFFER_SIZE = 4096;
+
+  private final Logger logger;
+  private final MethodHandle handle;
```

Comment:
You *could* play the same trick here with a single static handle that you create a bound version of at construction time.

Reply:
True ... I'll have a look.